pkg/apis/core/validation/validation_test.go

```diff
@@ -5273,6 +5273,204 @@ func TestPVCVolumeMode(t *testing.T) {
+					ReadOnly: true,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate on, new read-write and old read-write",
+			gateValue: true,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate off, new read-write and old read-write",
+			gateValue: false,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate on, new read-only and old read-only",
+			gateValue: true,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate off, new read-only and old read-only",
+			gateValue: false,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate on, new read-write and old read-only",
+			gateValue: true,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			expectError: false,
+		},
+		{
+			name:      "gate off, new read-write and old read-only",
+			gateValue: false,
+			volumes: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: false,
+				},
+			}},
+			oldVolume: []core.Volume{{
+				GCEPersistentDisk: &core.GCEPersistentDiskVolumeSource{
+					ReadOnly: true,
+				},
+			}},
+			expectError: true,
+		},
+	}
+	// enable gate
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			fidPath := field.NewPath("testField")
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SkipReadOnlyValidationGCE, testCase.gateValue)()
+			errs := ValidateReadOnlyPersistentDisks(testCase.volumes, testCase.oldVolume, fidPath)
+			if !testCase.expectError && len(errs) != 0 {
+				t.Errorf("expected success, got:%v", errs)
+			}
+		})
+	}
+	// disable gate
```

Comment:
I don';t see what is different in this block than the previous?  Is it a holdover from a previous draft?